ros/src/waypoint_follower/src/pure_pursuit_core.cpp

```diff
@@ -249,27 +249,42 @@ bool PurePursuit::verifyFollowing() const
     return false;
   }
 }
+
+////////////////////////////////////////////////
+// CHANGED THIS TO AVOID WANDERING OF THE CAR //
+////////////////////////////////////////////////
+//geometry_msgs::Twist PurePursuit::calcTwist(double curvature, double cmd_velocity) const
+//{
+//  // verify whether vehicle is following the path
+//  bool following_flag = verifyFollowing();
+//  static double prev_angular_velocity = 0;
+//
+//  geometry_msgs::Twist twist;
+//  twist.linear.x = cmd_velocity;
+//
+//  if (!following_flag)
+//  {
+//    //ROS_ERROR_STREAM("Not following");
+//    twist.angular.z = current_velocity_.twist.linear.x * curvature;
+//  }
+//  else
+//  {
+//    twist.angular.z = prev_angular_velocity;
+//  }
+//
+//  prev_angular_velocity = twist.angular.z;
+//  return twist;
+//}
 geometry_msgs::Twist PurePursuit::calcTwist(double curvature, double cmd_velocity) const
```

Comment:
Is this modified to follow the points all times?

Reply:
Yes. Without this change, the car was wabbling and behaved like it was controlled by a cautious but drunk driver. This change was suggested around 10:40 in DBW walkthrough. https://youtu.be/kdfXo6atphY?t=640

Reply:
This might have a different effect on the real car. Wouldn't it be better to have a low threshold (like Elgeweily's apprach) rather than updating twist.angular.z every time ?

Reply:
Addressed in #3.